### PR TITLE
feat: expose sketch specifics from supernode class

### PR DESCRIPTION
**Pull request by Victor-C-Zhang**
Distributing sketch queries needs to be able to get sketches from within a supernode. Right now, the necessary functions are private.

### CODE REVIEW

File: include/supernode.h
Comment (etwest) [inline include/supernode.h]: Is it possible to only make the const pointer public? This might require making Sketch write_binary const but that’s doable.

Comment (Victor-C-Zhang) [reply include/supernode.h]: I don't think that's how `const` member functions work. Afaik `const` versions of member functions are invoked only when the object is `const`. That means the `supernode` would need to be a `const` supernode, which clearly it cannot be since we are actively modifying it,

EDIT: Though now that I think about it, I could create a `get_curr_sketch()` function and just keep all native `get_sketch()` functions private.

Comment (Victor-C-Zhang) [reply include/supernode.h]: Why do you want this anyways?

Comment (etwest) [reply include/supernode.h]: I'm not saying the member function should be const but rather return a const pointer to a sketch object. The reason I want this is so that the caller of get_sketch can't muck with the internals of the supernode and make the data invalid for example.

Comment (etwest) [reply include/supernode.h]: Sketch const * is really what I want. Since we aren't returning a reference to the pointer we don't need to protect the pointer with const.

Comment (Victor-C-Zhang) [reply include/supernode.h]: Ok

Comment (Victor-C-Zhang) [reply include/supernode.h]: But this is impossible. We cannot have both a version of `get_sketch` that returns a const pointer and one that doesn't. Reason being that we need `Supernode` to be able to modify `Sketch` and we can't return 2 different things from the same method signature. Unless you really really want this compile-time safety measure, I don't really see how this would be beneficial for us.

Comment (etwest) [reply include/supernode.h]: Yeah... could also change the name of the public function. Maybe `read_sketch()` or something. I'll leave this up to you.

File: src/graph.cpp
Comment (etwest) [inline src/graph.cpp]: TODO: remove this change before merging

Comment (Victor-C-Zhang) [reply src/graph.cpp]: good catch

**Review by etwest — APPROVED** [review]
looks good

**Pull request merged**
